src/runner/runner.module.ts

```diff
@@ -1,23 +1,65 @@
-import { Module } from '@nestjs/common'
-import { BackupRunnerService } from './backup-runner/backup-runner.service'
-import { ChargerRunnerService } from './charger-runner/charger-runner.service'
-import { InvoiceRunnerService } from './invoice-runner/invoice-runner.service'
-import { OrderExpireRunnerService } from './order-expire-runner/order-expire-runner.service'
-import { PaymentRenewRunnerService } from './payment-renew-runner/payment-renew-runner.service'
-import { PorterRunnerService } from './porter-runner/porter-runner.service'
-import { PortfolioProjectExpireRunnerService } from './portfolio-project-expire-runner/portfolio-project-expire-runner.service'
-import { ReminderRunnerService } from './reminder-runner/reminder-runner.service'
+import { cwd } from 'process';
+import { DynamicModule, Module, Type } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { BullModule } from '@nestjs/bull';
+import { ScheduleModule } from '@nestjs/schedule';
+import { TypeOrmModule } from '@nestjs/typeorm';
 
-@Module({
-  providers: [
-    BackupRunnerService,
-    ChargerRunnerService,
-    InvoiceRunnerService,
-    OrderExpireRunnerService,
-    PaymentRenewRunnerService,
-    PorterRunnerService,
-    PortfolioProjectExpireRunnerService,
-    ReminderRunnerService,
-  ],
-})
-export class RunnerModule {}
+import { AppDataSourceConfig } from '~/data-source';
+import { LockModule } from '~/utility/lock/lock.module';
+import { Runner } from './runner';
+import { RunnerService } from './runner.service';
+import { ExampleRunner } from './example.runner';
+import { RunnerType } from './runner.type';
+
+@Module({})
+export class RunnerModule {
+  static createRunnerProvider(workerName: RunnerType): Type<Runner> {
+    switch (workerName) {
+      case RunnerType.EXAMPLE_RUNNER:
+        return ExampleRunner;
+      default:
+        throw new Error(`Given runner name is not exists: ${workerName}`);
+    }
+  }
```

Comment:
it is annoying to add both RunnerType and here while adding a new runner. i think it is easy to forget to add both side. we can merge the switch case into enum. plz see the `runner.type`